clang/include/clang/Sema/AnalysisBasedWarnings.h

```diff
@@ -57,6 +61,8 @@ class AnalysisBasedWarnings {
 
   enum VisitFlag { NotVisited = 0, Visited = 1, Pending = 2 };
   llvm::DenseMap<const FunctionDecl*, VisitFlag> VisitedFD;
+  llvm::MapVector<VarDecl *, SmallVector<PossiblyUnreachableDiag, 4>>
+      VarDeclPossiblyUnreachableDiags;
```

Comment:
I feel like a `std::multimap<VarDecl *,  PossiblyUnreachableDiag>` would be a more suited container here

(still preserve order and avoid having lots of SmallVector)

Reply:
Hi, this is done in [`c6ec954` (#163885)](https://github.com/llvm/llvm-project/pull/163885/commits/c6ec954bf777b812848774baf61f0e74a922d1d1) but I think it wasn't as elegant as I was expecting. This is because `SmallVector<PossiblyUnreachableDiag, 4> PossiblyUnreachableDiags;` in `ScopeInfo.h` is already declared as a vector. The change meant I needed to template `emitPossiblyUnreachableDiags` and use `llvm::make_second_range`.